services/tempfile.c

```diff
@@ -88,10 +88,10 @@ int dcc_make_tmpnam(const char *prefix, const char *suffix, char **name_ret, int
 #endif
 
     do {
-        if (snprintf(tmpname, tmpname_length, "%s/%s_%08lx%s",
+        if (snprintf(tmpname, tmpname_length, "%s/%s_%08lu%s",
                      (relative ? _PATH_TMP + 1 : _PATH_TMP),
                      prefix,
-                     random_bits & 0xffffffffUL,
+                     random_bits & ULONG_MAX,
```

Comment:
The number is not supposed to be a "random" upper limit, it's supposed to be all binary ones, so the original is in fact better.